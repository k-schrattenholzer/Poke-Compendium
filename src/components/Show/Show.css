.Show {
  border-radius: 20%;
  padding: 15px 25px;
  margin: 10px 10px;
  height: 420px;
  width: 400px;
  border: 1px solid white;
  display: flex;
  flex-direction: column;
  align-content: flex-start;
  font-size: 14px;
}

.Show >* {
  word-wrap: break-word;
  overflow: hidden;
  text-overflow: clip;

}

.Show >a >img {
  height: 150px;
  border-radius: 20%;
  padding: 1em;
}