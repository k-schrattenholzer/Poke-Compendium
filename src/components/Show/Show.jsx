
import { removeTags } from '../../utils/utils.js';
import './Show.css';

export default function Show({ show }) {
//destructure props needed from show
const { id, name, lang, premiere, image, summary, url } = show;


const hasImage = image !== null

const summaryString = removeTags(summary);

  return (
    <article className='Show' key={id}>
      <a href={url}>
        { hasImage && <img src={image} alt={id}/>}
        
      </a>
      <h4>{name}</h4>
      <ul>
        <li key={id*5} ><b>language</b> {lang}</li>
        <li key={id*6}>premiere: {premiere}</li>
      </ul>
      <div>{summaryString}</div>
    
    </article>
  )
}
