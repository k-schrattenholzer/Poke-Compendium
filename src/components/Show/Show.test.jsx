import { screen, render } from "@testing-library/react";
import Show from './Show.jsx';

it('should render one show component', () => {
  render (
    <Show show={{
      id: 6,
    name:'What We Do in the Shadows',
    lang: "English",
    premiere: '01*01*01',
    image: 'https://assets.fxnetworks.com/cms/prod/2021/07/22/web_poster_540x796.jpg',
    summary: 'a rag tag group learns the meaning of christmas',
    url: 'https://assets.fxnetworks.com/cms/prod/2021/07/22/web_poster_540x796.jpg',
    genres: ['horror', 'thriller', 'family']
    }}/>
  )
})