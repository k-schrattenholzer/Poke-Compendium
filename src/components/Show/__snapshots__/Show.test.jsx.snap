// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render one show component 1`] = `
<h4>
  What We Do in the Shadows
</h4>
`;
